torch/csrc/jit/codegen/cuda/ir_utils.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/type.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+class Statement;
+
+namespace ir_utils {
+
+template <
+    typename FilterType,
+    template <typename, typename...> typename Container,
+    typename ElementType>
+Container<FilterType*> filterVals(const Container<ElementType*>& container) {
+  Container<FilterType*> out;
+  for (auto& s : container) {
```

Comment:
nit: using the explicit root type here woudl avoid the need for the `template as<FilterType>` below

Reply:
Sorry, can you give me more details?

Reply:
I just meant `for (Statement* s : container) ...`.